typescript/infra/test/core.test.ts

```diff
@@ -100,6 +100,36 @@ describe('core', async () => {
     });
   });
 
+  describe('proxy upgrades', async () => {
+    beforeEach(async () => {
+      deployer = new HyperlaneCoreInfraDeployer(
+        multiProvider,
+        coreConfig,
+        environment,
+      );
+      await deployer.deploy();
+    });
+
+    it('deploys a new implementation if it has been removed from the artifacts', async () => {
```

Comment:
This approach has been contentious, so I'm happy to remove this (& also the changes to `isProxyAddresses` [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1765/files#diff-c9d6f33c59f182d547b3785c9c2a7aae7f46c529178ab6c3e7372eb953402f3aR20)) if you think that's best @yorhodes